examples/mfem/bp1.okl

```diff
@@ -18,46 +18,48 @@
 typedef int CeedInt;
 typedef double CeedScalar;
 
-// OCCA parser doesn't like __global here
-typedef /*__global*/ double gCeedScalar;
-
-// OCCA parser doesn't like struct here
-//typedef /*__global*/ struct BuildContext gBuildContext;
-
 // *****************************************************************************
-// OCCA parser doesn't like struct here either
+// OCCA parser doesn't like struct here
 //struct BuildContext { CeedInt dim, space_dim; };
 
 // *****************************************************************************
 @kernel void f_build_mass(int *ctx, CeedInt Q,
                           const int *iOf7, const int *oOf7,
                           const CeedScalar *in, CeedScalar *out) {
-  const CeedScalar
-    *J = in + iOf7[0],
-    *qw = in + iOf7[1];
-  CeedScalar
-    *rho = out + oOf7[0];
+//  const CeedScalar
+//    *J = in + iOf7[0],
+//    *qw = in + iOf7[1];
+//  CeedScalar
+//    *rho = out + oOf7[0];
   const int dim = ctx[0];
   const int space_dim = ctx[1];
   const CeedInt dims = dim + 10*space_dim;
   if (dims == 11) {
     for (int i=0; i<Q; i++; @tile(TILE_SIZE,@outer,@inner)) {
-      rho[i] = J[i] * qw[i];
+      out[oOf7[0]+i] = in[iOf7[0]+i] * in[iOf7[1]+i];
     }
   } else if (dims == 22) {
    for (int i=0; i<Q; i++; @tile(TILE_SIZE,@outer,@inner)) {
      // 0 2
      // 1 3
-     rho[i] = (J[i+Q*0]*J[i+Q*3] - J[i+Q*1]*J[i+Q*2]) * qw[i];
+     out[oOf7[0]+i] = (in[iOf7[0]+i+Q*0]*in[iOf7[0]+i+Q*3] -
+                       in[iOf7[0]+i+Q*1]*in[iOf7[0]+i+Q*2]) * in[iOf7[1]+i];
     }
   } else if (dims == 33) {
     for (int i=0; i<Q; i++; @tile(TILE_SIZE,@outer,@inner)) {
       // 0 3 6
       // 1 4 7
       // 2 5 8
-      rho[i] = (J[i+Q*0]*(J[i+Q*4]*J[i+Q*8] - J[i+Q*5]*J[i+Q*7]) -
-                J[i+Q*1]*(J[i+Q*3]*J[i+Q*8] - J[i+Q*5]*J[i+Q*6]) +
-                J[i+Q*2]*(J[i+Q*3]*J[i+Q*7] - J[i+Q*4]*J[i+Q*6])) * qw[i];
+      out[oOf7[0]+i] = (in[iOf7[0]+i+Q*0]*
+                          (in[iOf7[0]+i+Q*4]*in[iOf7[0]+i+Q*8] -
+                           in[iOf7[0]+i+Q*5]*in[iOf7[0]+i+Q*7]) -
+                        in[iOf7[0]+i+Q*1]*
+                          (in[iOf7[0]+i+Q*3]*in[iOf7[0]+i+Q*8] -
+                           in[iOf7[0]+i+Q*5]*in[iOf7[0]+i+Q*6]) +
+                        in[iOf7[0]+i+Q*2]*
+                          (in[iOf7[0]+i+Q*3]*in[iOf7[0]+i+Q*7] -
+                           in[iOf7[0]+i+Q*4]*in[iOf7[0]+i+Q*6])) *
+                        in[iOf7[1]+i];
```

Comment:
@dmed256 What's the timeframe for the preferred syntax to be supported in Occa for OpenCL targets? Is there a specific Occa issue for us to watch? Not being able to name an array is pretty sad for maintainability.

Reply:
@jedbrown https://github.com/libocca/occa/issues/86 and https://github.com/libocca/occa/issues/166 need to be resolved. I can get these done before August is over, so about 2 weeks